tests/trainer/test_trainer.py

```diff
@@ -637,23 +637,26 @@ def test_trainer_max_steps_accumulate_batches(tmpdir):
     assert trainer.global_step == trainer.max_steps, "Model did not stop at max_steps"
 
 
-def test_benchmark_option(tmpdir):
+@pytest.mark.parametrize(
+    ["benchmark", "deterministic", "expected"],
+    [
+        (None, False, True),
+        (None, True, False),
+        (True, False, True),
+        (True, True, True),
+        (False, True, False),
+        (False, False, False),
+    ],
+)
+def test_benchmark_option(benchmark, deterministic, expected):
     """Verify benchmark option."""
 
-    model = BoringModel()
-
-    # verify torch.backends.cudnn.benchmark is not turned on
-    assert not torch.backends.cudnn.benchmark
+    original_val = torch.backends.cudnn.benchmark
 
-    # fit model
-    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, benchmark=True)
-    trainer.fit(model)
-
-    # verify training completed
-    assert trainer.state.finished, f"Training failed with {trainer.state}"
+    trainer = Trainer(benchmark=benchmark, deterministic=deterministic)
```

Comment:
This should silence the pre-commit error in CI

```suggestion
    _ = Trainer(benchmark=benchmark, deterministic=deterministic)
```